sd: mmc: Remove unwanted request to card for reading OCR content

AlvinPaul26 opened the pull request:
As part of the MMC card initialization sequence, CMD1 command send multiple times to the card, first time to identify the card type and second time to check the requested voltage window is supported or not. There is a chance that after issuing the CMD1 for first time, card will move to the ready state before issuing the CMD1 for second time. In this case, card will not respond to the CMD1 send for the second time as card is already moved to ready state and this leads to failures. Hence remove the separate card identification logic and call mmc_send_op_cond only once to check both supported voltage window and card type identification.

decsny reviewed (dismissed):
seems fine to me

danieldegrasse reviewed (commented):
Looking at the spec I agree the second CMD1 could cause issues. We should drop the `LOG_ERR` (or reduce it to a `LOG_DBG`) on line 130 though- I suppose that since MMC cards are queried last, this isn't really a big issue (as working SDIO and SDMMC cards should never make it to this code flow). I just would prefer that SDIO/SDMMC cards that fail initialization don't also have a "Failed to query card OCR" log from the MMC code included in the output

AlvinPaul26 commented:
yeah, I will better change it into LOG_DBG.
thanks.

AlvinPaul26 committed:
sd: mmc: Remove unwanted request to card for reading OCR content

As part of the MMC card initialization sequence, CMD1 command send
multiple times to the card, first time to identify the card type and
second time to check the requested voltage window is supported or not.
There is a chance that after issuing the CMD1 for first time, card will
move to the ready state before issuing the CMD1 for second time. In this
case, card will not respond to the CMD1 send for the second time as card
is already moved to ready state and this leads to failures. Hence remove
the separate card identification logic and call mmc_send_op_cond only
once to check both supported voltage window and card type
identification.

Signed-off-by: Paul Alvin <alvin.paulp@amd.com>
